### ISSUE TITLE
Support streaming API

### ISSUE THREAD

**Issue by tebeka** [tebeka/go2xunit#41]
Some tools (like TeamCity) can get report testing results in streaming manner. Support that in the library. See #39  for a discussion.

**Comment by tebeka** [comment]
My current thought is to break the lexer out and have it report tokens (start test, data, end test ...) via a channel.

**Comment by tebeka** [comment]
Just merged streaming branch to master with initial gotest lexer. @tamird can you have a look at see if this works for you?

**Comment by tamird** [comment]
Thanks. Unfortunately it seems that I need to write the parser, still? That's kind of what I was hoping to avoid :(

**Comment by tebeka** [comment]
Parser is on the way to be written (see streaming branch). However a parser will be able to emit one test at a time, is this what you wanted?

**Comment by tamird** [comment]
Yeah - one test at a time is great (though you may need to do a bit of
buffering to address #45).
